Add several features

babky-atteq opened the pull request:
The added features mainly include:

* Add possibility to generate config and use it with a different Sphinx instance. 
* Do not start Sphinx instance during the application startup. 
* Add possibility to set up the Sphinx binary path.
* Add support for unit tests in the application using django-sphinxql using the local instance.

babky committed:
Add a possibility to connect to other hosts than local

Nowadays sphinxql can connect to hosts specified in connection_params in
the configuration. We can connect to remote sphinx servers. With this
the necessity for generating sphinx.conf during startup is void.
Therefore this functionality is refactored out and a command to do this
is added (generate_shpinx_conf). Also the directories required to place
the file are created as well.

babky committed:
Add a command to generate sphinx.conf

babky committed:
Perform maintenance and changes regarding sphinx startup.

The following tasks were done:

1. Allow the connection to external search daemon to be configured.
2. Run the sphinx on background for unit tests.
3. Do not start sphinx by default - connect to the given instance.
4. Refactor SpinxQuerySet.
 + Do not allow querying with sphinx and django ordering, raise
   NotImplementedError.
5. Reflect the changes in tests.

babky committed:
Improve index and start commands

The improvement lies in better output processing - directly to stdout
if required.

babky committed:
Remove warnings from configuration - charset_type and address

The two mentioned settings are no longer available.

babky committed:
Make index_sphinx command to be able to be run without --rotate option

babky committed:
Merge branch 'master' into connection-configuration

babky committed:
Merge pull request #1 from AtteqCom/connection-configuration

Connection configuration

babky committed:
Remove unused constant.

babky committed:
Merge remote-tracking branch 'origin/master'

babky committed:
Improve log output when configuration exception occurs

babky committed:
Configuration for uwsgi.

babky committed:
Fix the configuration host/port parsing and the unit tests

babky committed:
Add test support for other django projects.

SphinxQlTestCase is added with the setUp and tearDown methods so that
we can start/stop/index sphinx instance on demand.

babky committed:
Return the outputs of index and reindex in unit tests.

babky committed:
Add a possibility to tweak unit test settings using a local settings

babky committed:
Improve documentation and readme

jorgecarleitao commented:
Hey. Thanks a lot for this, @babky-atteq. I have not checked specifically what you did, but the tests are not passing. Let me know when you are done so I can review it and merge accordingly.

ghost commented:
Hello, I'll fix them in a couple of days. They were passing on my computer, and I was not clever enough to check them on Travis. But I'll make them working.

babky committed:
Update README.md

Describe how unit test settings can be adapted to a local environment.

babky committed:
Update README.md

Describe sphinx_bin_path usage.

babky committed:
Fix the unit tests

The library was not working with Python 3.4, the compatibility was
added. Also the unit tests now work on slower computers - a limit
of 2 seconds was added to wait until the searchd starts. This can
be further improved to detect a real start of searchd.

babky committed:
Add typing to requirements for travis.

babky committed:
Create databases for unit tests. Test the tests.

babky committed:
Run the tests for all the environments.

babky committed:
Update README.md to use this repository build and coverage status

babky committed:
Try avoiding losing connection to MySQL during travis-ci build

ghost commented:
Hello @jorgecarleitao, I've made the tests passing on Travis. You can now check the request and merge if you like.

Notice that I slightly changed the behavior in the way how the sorting is defined. When both sorting in Sphinx and in Django is used an exception is raised so that there's no default behavior or ignoring of one way of sorting. There should be no other interface/contract break.

The coverage drop is because of the added commands to generate sphinx configuration.

Aspect26 committed:
Update README.md

Correct command names

babky committed:
Merge pull request #2 from AtteqCom/Aspect26-patch-readme

Update README.md

babky committed:
Allow sphinxql to use `db_table` from Meta in the index definition

babky committed:
Merge remote-tracking branch 'origin/master'

ghost commented:
We also added a possibility to define the table/index name via 

```
class Index(indexes.Index):
    class Meta:
        db_table= '<index_name>'
```

jorgecarleitao commented:
I have not forgotten this. It is very solid work. I would like to review it before pushing to the master, and will probably make the history linear. Regardless, great work @babky-atteq

babky committed:
Fix running unit tests without pymysql

babky committed:
Improve the testing

babky committed:
Split test case into mixing and an actual test case

babky committed:
Merge branch 'master' of https://github.com/AtteqCom/django-sphinxql

geoolekom commented:
What about this feature? Can you merge it?

babky commented:
^^ @jorgecarleitao: we would also be happy to use a pypi package instead of github repo reference :+1: If required, we could make some improvements.
